tasks/CommandInjection.cpp

```diff
@@ -0,0 +1,83 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "CommandInjection.hpp"
+
+using namespace auv_control;
+
+CommandInjection::CommandInjection(std::string const& name)
+    : CommandInjectionBase(name)
+{
+}
+
+CommandInjection::CommandInjection(std::string const& name, RTT::ExecutionEngine* engine)
+    : CommandInjectionBase(name, engine)
+{
+}
+
+CommandInjection::~CommandInjection()
+{
+}
+
+bool CommandInjection::calcOutput()
+{
+    base::LinearAngular6DCommand output_command = merged_command;
+
+    if(_cmd_injection.connected())
+    {
+        base::LinearAngular6DCommand cmd_injection;
+        if(_cmd_injection.readNewest(cmd_injection) == RTT::NewData)
```

Comment:
> You have no guarantee that there is a data read here since you don't test with RTT::OldData. 

I have it implicit by checking the timeout. But I will add it.

> Moreover, don't use readNewest. It's really unnecessary, and is problematic when running a component against log streams.

I would assume to get the latest sample would be exactly what I want to have here. Especially when replaying log data faster than the update hook is triggered.

> Moreover deeply, imbricated ifs() are really bad form. Avoid them as much as you can.

True, but since this happens in a calcOutput() callback that in this case always should write out a sample and return true it will be harder to read and understand the code.